src/json.hpp

```diff
@@ -10908,6 +10924,42 @@ class basic_json
         return result;
     }
 
+#ifdef JSON_HAS_CPP_14
+    /*!
+    @brief find an element in a JSON object
+    @copydoc find(typename object_t::key_type)
+    */
+    template<typename KeyT>
+    iterator find(KeyT&& key)
```

Comment:
Could we define those perfect-forwarded overloads for C++11 as well?

Reply:
The templated member functions don't exist in 11. They were added in 14.

Reply:
Yes, but we could keep those overloads in C++11, they will call the regular comparators which will perform implicit conversions, as for:

```cpp
std::cout << std::multiplies<int>{}(2.0f, 2.0f) << std::endl;
```

I haven't tested my suggestion though, so it might very well be bogus.

Reply:
My first though was that it might be weird for users that aren't familiar with the difference in capabilities between `std::map` in 11 and 14 to have the templated type that only works with things implicitly convertible to the key type, but maybe that's not an issue, as you'd get the same issue with anything that isn't comparable with the key type with the 14 version.  The STL kept the non-template `find()` and `count()`, but likely only for backwards compat.  If it works, I think it would be better with just the templated versions.  If we can't eliminate the fixed type versions, we should add `const &` to them to match the `std::map` versions.  Right now we make a copy unnecessarily.  (I just noticed this now.)

Reply:
I like the suggestion, pushed changes to just have perfect-forwarding only